airbyte-ci/connectors/pipelines/tests/test_changelog.py

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+import difflib
+import json
+import random
+import shutil
+from pathlib import Path
+from typing import List
+from unittest.mock import AsyncMock, MagicMock
+
+import anyio
+import pytest
+import semver
+from connector_ops.utils import Connector, ConnectorLanguage
+from dagger import Directory
+from pipelines import main_logger
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+from pipelines.airbyte_ci.connectors.publish import pipeline as publish_pipeline
+from pipelines.airbyte_ci.connectors.upgrade_cdk import pipeline as upgrade_cdk_pipeline
+from pipelines.helpers.changelog import Changelog
+from pipelines.models.steps import StepStatus
+
+pytestmark = [
+    pytest.mark.anyio,
+]
+
+PATH_TO_INITIAL_FILES = Path("airbyte-ci/connectors/pipelines/tests/test_changelog/initial_files")
+PATH_TO_RESULT_FILES = Path("airbyte-ci/connectors/pipelines/tests/test_changelog/result_files")
+WRITE_TO_RESULT_FILE = False
+
+
+def check_result(changelog: Changelog, result_filename: str):
+    markdown = changelog.to_markdown()
+    result_filepath = PATH_TO_RESULT_FILES / result_filename
+    if not result_filepath.exists():
+        expected_text = ""
+    else:
+        expected_text = result_filepath.read_text()
+    diff = "".join(difflib.unified_diff(expected_text.splitlines(1), markdown.splitlines(1)))
+    if WRITE_TO_RESULT_FILE:
+        result_file = open(result_filepath, "w")
+        result_file.write(markdown)
+        result_file.close()
+    assert diff == ""
+
+
+@pytest.mark.parametrize("filename", ["valid_changelog_at_end.md", "valid_changelog_in_middle.md"])
+def test_single_insert(dagger_client, tmp_path, filename):
```

Comment:
You're not using the `tmp_path` fixture in these tests if I'm not mistaken, please remove the parameter.